### ISSUE TITLE
no-unused-variable doesn't work in declaration files

### ISSUE THREAD

**Issue by karol-majewski** [palantir/tslint#3883]
### Bug Report

- __TSLint version__: 5.10.0
- __TypeScript version__: 2.8.3
- __Running TSLint via__: CLI

##### `a.ts`

```ts
export const name = "John";
export const surname = "Doe";
```

##### `b.d.ts`

```ts
import { name, surname } from "./a";

export { name };
```

with `tslint.json` configuration:

```json
{
  "extends": [
    "tslint:latest"
  ],
  "rules": {
    "no-unused-variable": true
  }
}

```

#### Actual behavior

No error.

#### Expected behavior

Error: `'surname' is declared but its value is never read.`

**Comment by giladgray** [comment]
@karol-majewski what happens in the original `b.ts` source file? does it use `surname` in a way that is not preserved in the output? if not, then this behavior is expected.

**Comment by giladgray** [comment]
basically, this behavior looks correct. TSLint treats .d.ts files as source code unless you exclude/ignore them. so the lint rule is catching a valid error.

**Comment by karol-majewski** [comment]
@giladgray It does not use it in any way.

This becomes problematic when you import and export from & to `.d.ts` files (your application domain model for example) and you end up with dead code pilling up, with no way to manage it.

**Comment by JoshuaKGoldberg** [comment]
Amusing: this is the opposite of requests such as #190 and #434. Imports do seem reasonable to not exclude for being declarations.

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #4534. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
